daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala

```diff
@@ -864,29 +876,42 @@ abstract class TestCase(testCaseXML: NodeSeq, val parent: DFDLTestSuite) {
         nBits,
         optExpectedErrors,
         optExpectedWarnings,
+        optExpectedLogs,
         optExpectedValidationErrors,
         validationMode,
         roundTrip,
-        implString)
+        implString,
+        parent.tdmlLogger)
 
     }
+
+    if (parent.tdmlLogger ne null)
+      parent.tdmlLogger.clearDiagnostics()
```

Comment:
This `clearDiagnostics` thing is sort of an artifact of our logger and the way it works (i.e. dealing with global state). It concerns me that other logging systems might need very different abstractions that our basic appended might not support. I wonder if there's a good way to make it so that TDMLProcessor implementations do not need to be restricted to a specific logger abstraction?

For example, what if creating/configuring the logging mechanism is handled entirely within the TDMLProcessor implementation? And the TDML Runner has no concept of that logging mechanism? Then TDML Processor can capture logs however it wants and add them to the CompileResult/ParseResult/UnaprseResult. Then the TDMLRunner can then access them with a new function (e.g. `getLogs`) and do the comparisons you do. This way, the only thign the TDMLRunner needs to know is there is a way to get logs from a result, and it's entirely up to the processor how to best do that.

An added benefit is you don't need to pass around the tdmlLogger to a bunch of functions, since it's all contained in the TDML Processor and the Parse/UnparseResults, which we already pass around where needed.